### PR TITLE
[IMP] bus: use lazy start for event dispatcher + allow to peek bus

### PR DESCRIPTION
**Pull request by amigrave**
Use lazy start for the bus dispatcher.
Also allow to peek the bus for notifications and listening channels

### CODE REVIEW

File: addons/bus/models/bus.py [hunk 1]
```diff
@@ -178,15 +184,16 @@ def start(self):
...
-            # disabled in prefork mode
-            return
         else:
             # threaded mode
             self.Event = threading.Event
             t = threading.Thread(name="%s.Bus" % __name__, target=self.run)
             t.daemon = True
             t.start()
+        self.started = True
         return self
```
Comment (antonylesuisse) [inline addons/bus/models/bus.py]: Nope

File: addons/bus/models/bus.py [hunk 2]
```diff
@@ -178,15 +184,16 @@ def start(self):
             import gevent
             self.Event = gevent.event.Event
             gevent.spawn(self.run)
-        elif odoo.multi_process:
```
Comment (antonylesuisse) [inline addons/bus/models/bus.py]: nope

**Comment by amigrave** [comment]
@antonylesuisse I guess you mean if one would try to call `_poll()` directly in multiprocess mode without passing by controller's `poll()` ? Indeed in such a case dispatcher would be None.

So do you mean I should handle this case or shall I just remove this `dispatcher = None` in multiprocess mode altogether ?

**Comment by amigrave** [comment]
@KangOl the changes reflects my discussion with @antonylesuisse but I think it's still not good. I have to catch him and talk about that.

**Comment by amigrave** [comment]
> I would have separate the peeking option in its own commit, but ok with the changes.

@KangOl done!

**Comment by amigrave** [comment]
Closed by a2ed3d3d5bdb6025a1ba14ad557a115a86413e65 and f2f99dc091947c0799b17c767a6ad923cd535025

**Pull request closed without merging**